scrapper.py

```diff
@@ -31,67 +42,260 @@ class Crawler:
 
     def _fill_article_with_meta_information(self, article_soup):
-        pass
+        try:
+            title = article_soup.title.text
+            self.article.title = title
+
+            credit = article_soup.find('div', {'class': 'credits t-caption'}).text.strip().split('\n')[0]
+            if 'Автор:' in credit:
+                author = article_soup.find('div', {'class': 'credits t-caption'}).text.strip().split('\n')[0][7:]
+            elif 'Источник:' in credit:
+                author = article_soup.find('div', {'class': 'credits t-caption'}).text.strip()
+                author = author.split('\n')[0][9:].strip()
+            else:
+                author = 'Not found'
+            self.article.author = author
+            when = article_soup.find('div', {'class': 'b-caption'}).text.strip().split('\n')[1]
+            self.article.date = self.unify_date_format(when)
+
+            topic = article_soup.find('div', {'class': 'b-caption'}).text.strip().split('\n')[0]
+            self.article.topics = topic
+        except AttributeError:
+            print('    something is off with', self.full_url)
+        # print(title)
+        # self.article.title = title
 
     @staticmethod
     def unify_date_format(date_str):
         """
         Unifies date format
         """
-        pass
+        day, month, year = date_str.split()
+        if len(day) == 1:
+            day = '0' + day
+        match = {'янв': '01',
+                 'фев': '02',
+                 'мар': '03',
+                 'апр': '04',
+                 'май': '05',
+                 'июн': '06',
+                 'июл': '07',
+                 'авг': '08',
+                 'сен': '09',
+                 'окт': '10',
+                 'ноя': '11',
+                 'дек': '12'}
+        return date.fromisoformat(year + '-' + match[month] + '-' + day)
 
     def parse(self):
         """
         Parses each article
         """
-        pass
+        self.article.url = self.full_url
+        self.article.article_id = self.article_id
+        html = requests.get(self.full_url, 'html.parser').text
+        article_bs = BeautifulSoup(html, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        self.article.save_raw()
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if not os.path.exists(base_path):
+        os.makedirs(base_path)
+
+
+def enable_backup(base_path):
+    """
+    Creates folder for backup links if not created
+    """
+    if not os.path.exists(base_path):
+        os.makedirs(base_path)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as crawler_config:
+        config = json.load(crawler_config)
+    try:
+        good_response = list(map(lambda link: requests.get(link).status_code == 200,
```

Comment:
urls should not be checked by calling them. we need to be very accurate and minimize number of external requests